src/NUnitFramework/framework/Internal/ExceptionHelper.cs

```diff
@@ -33,6 +34,41 @@ namespace NUnit.Framework.Internal
     /// </summary>
     public class ExceptionHelper
     {
+#if !NET_4_5 && !PORTABLE
+        private static readonly Action<Exception> PreserveStackTrace;
+
+        static ExceptionHelper()
+        {
+            var method = typeof(Exception).GetMethod("InternalPreserveStackTrace", BindingFlags.Instance | BindingFlags.NonPublic);
+
+            if (method != null)
+            {
+                try
+                {
+                    PreserveStackTrace = (Action<Exception>)Delegate.CreateDelegate(typeof(Action<Exception>), method);
+                    return;
+                }
+                catch (InvalidOperationException) { }
+            }
+            PreserveStackTrace = _ => { };
+        }
+#endif
+
+        /// <summary>
+        /// Rethrows an exception, preserving its stack trace
+        /// </summary>
+        /// <param name="exception">The exception to rethrow</param>
+        public static void Rethrow(Exception exception)
```

Comment:
@thomaslevesque While I won't blame anybody for using it, I will warn them that it's subject to change. I've been doing it for so long that I tend to assume everyone knows it, but of course they don't. Shifting such things to internal is a good move. If it breaks something, at least it will break it in an obvious way, which is much better than what happens if we make a subtle change to the semantics of a method.

We have always stressed that there is a difference between public methods and public APIs. We'll continue to do that and try to give it better documentation where it's necessary.